src/couch/src/couch_httpd_auth.erl

```diff
@@ -186,6 +188,26 @@ proxy_auth_user(Req) ->
             end
     end.
 
+jwt_authentication_handler(Req) ->
+    Secret = config:get("couch_httpd_auth", "jwt_secret", ""),
+    UserClaim = config:get("couch_httpd_auth", "jwt_user_claim", "sub"),
+
+    case header_value(Req, "Authorization") of
+        "Bearer " ++ Jwt ->
+            case jwt:decode(?l2b(Jwt), ?l2b(Secret)) of
+                {ok, Claims} ->
+                    User = maps:get(?l2b(UserClaim), Claims),
+
+                    Req#httpd{user_ctx=#user_ctx{
+                        name=User,
+                        roles=[]
```

Comment:
this line can be removed (I realise it's mine) for clarity that JWT auth does not set roles.